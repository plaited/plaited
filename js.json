{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "plaited",
    "paths": {
      "plaited/jsx-runtime": [
        "./src/jsx/runtime.ts"
      ],
      "plaited/jsx-dev-runtime": [
        "./src/jsx/dev-runtime.ts"
      ]
    }
  }
}