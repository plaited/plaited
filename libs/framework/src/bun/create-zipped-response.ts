export const jsMimeTypes = ['text/javascript;charset=utf-8', 'application/json;charset=utf-8'] as const

export const isJsMimeType = (type: string): type is (typeof jsMimeTypes)[number] =>
  jsMimeTypes.includes(type as (typeof jsMimeTypes)[number])

export const createZippedResponse = (str: string, type: (typeof jsMimeTypes)[number]) => {
  const compressed = Bun.gzipSync(Buffer.from(str))
  return new Response(compressed, {
    headers: {
      'content-type': type,
      'content-encoding': 'gzip',
    },
  })
}
