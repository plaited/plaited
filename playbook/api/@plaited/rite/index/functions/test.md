**@plaited/rite** ( [Readme](../../README.md) \| API )

***

[Plaited Typedocs](../../../../modules.md) / [@plaited/rite](../../modules.md) / [index](../README.md) / test

# Function: test()

> **test**(`name`, `cb`): `void`

## Parameters

▪ **name**: `string`

▪ **cb**: `TestCallback`

## Returns

`void`

## Source

[libs/rite/src/test.ts:10](https://github.com/plaited/plaited/blob/0d4801d/libs/rite/src/test.ts#L10)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
