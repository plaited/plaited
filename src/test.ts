export * from './test/assert.types.js'
export * from './test/assert.constants.js'
export { assert } from './test/assert.js'
export { findByAttribute } from './test/find-by-attribute.js'
export { findByText } from './test/find-by-text.js'
export { fireEvent } from './test/fire-event.js'
export { match } from './test/match.js'
export { throws } from './test/throws.js'
export { wait } from './utils/wait.js'
export { AssertionError, MissingTestParamsError, TimeoutError } from './test/errors.js'
export { usePlay } from './test/use-play.js'
