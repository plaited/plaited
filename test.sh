#!/bin/bash

# Run yarn
yarn install --immutable

# Run yarn build
yarn build

# Run test
yarn test
