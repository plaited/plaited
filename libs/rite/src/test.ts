/// <reference lib="dom" />
/// <reference lib="dom.iterable" />

import { TestResultError, TestResult, BrowserSessionResult } from '@web/test-runner-core/browser/session.js'
import { assert, Assertion, AssertionError } from './assert.js'

type TestCallback = (t: Assertion) => Promise<void> | void

interface Test {
  (name: string, cb: TestCallback): void
  skip: (name: string, cb: TestCallback) => void
}

type TestCase = () => Promise<boolean> | boolean

class TestRunner {
  #completedTests = 0
  #failedTests = 0
  #errors: TestResultError[] = []
  #cases: TestCase[] = []
  #timeout = 5_000
  #tests: TestResult[] = []
  updateTimeout(time: number) {
    this.#timeout = time
  }
  get timeout() {
    return this.#timeout
  }
  addCase(test: TestCase) {
    this.#cases.push(test)
  }
  addResult(result: TestResult) {
    this.#tests.push(result)
  }
  addError(obj: TestResultError) {
    this.#errors.push(obj)
  }
  async run(name: string) {
    for (const testCase of this.#cases) {
      try {
        const passed = await testCase()
        this.#completedTests++
        if (!passed) {
          this.#failedTests++
        }
      } catch (error) {
        console.error(error)
      }
    }

    if (this.#completedTests === this.#cases.length) {
      const results: BrowserSessionResult = {
        passed: !this.#failedTests,
        testResults: {
          name,
          suites: [],
          tests: this.#tests,
        },
        errors: this.#errors,
      }
      return results
    } else {
      throw new AssertionError(`Not all test completed`)
    }
  }
}

declare global {
  interface Window {
    __rite_test_runner?: TestRunner
  }
}

window.__rite_test_runner = new TestRunner()

const throwTimeoutError = async () => {
  const timeout = window.__rite_test_runner.timeout
  await assert.wait(timeout)
  throw new Error(`test takes longer than ${timeout}ms to complete`)
}

export const test: Test = (name, callback) => {
  window.__rite_test_runner.addCase(async () => {
    const startTime = performance.now()
    const result: TestResult = {
      name,
      passed: true,
      skipped: false,
    }
    try {
      await Promise.race([callback(assert), throwTimeoutError()])
      const msg = `✓ ${name}`
      // eslint-disable-next-line no-console
      console.log(msg)
    } catch (error) {
      const msg = `✗ ${name}`
      // eslint-disable-next-line no-console
      console.log(msg)
      result.passed = false
      if (error instanceof AssertionError) {
        const obj: Pick<TestResultError, 'actual' | 'expected' | 'message'> = JSON.parse(error.message)
        result.error = { name, stack: `${error.stack}`, ...obj }
        window.__rite_test_runner.addError({ name, ...obj })
      } else {
        throw new Error(error)
      }
    }
    result.duration = performance.now() - startTime
    window.__rite_test_runner.addResult(result)
    return result.passed
  })
}
test.skip = (name: string, _: TestCallback) => {
  window.__rite_test_runner.addCase(() => {
    const result = {
      name,
      passed: true,
      skipped: true,
    }
    window.__rite_test_runner.addResult(result)
    return true
  })
}
