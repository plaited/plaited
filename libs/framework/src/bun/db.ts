import { Database } from "bun:sqlite";

export const db = new Database(":memory:");
db.query(`create table templates;`);