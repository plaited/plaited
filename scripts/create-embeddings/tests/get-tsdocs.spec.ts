import { describe, expect, test } from 'bun:test'
import { getTSDocs } from '../get-tsdoc-ast.js'

describe('getTSDocs', () => {
  test('should generate AST documentation structure', () => {
    const source = `
      /**
       * Adds two numbers together
       * @param a - First number
       * @param b - Second number
       * @returns The sum of a and b
       * @example
       * \`\`\`ts
       * add(1, 2) // returns 3
       * \`\`\`
       */
      function add(a: number, b: number): number {
        return a + b;
      }
    `
    const docs = getTSDocs(source)
    expect(docs).toMatchSnapshot()
  })

  test('should handle functions with deprecated tag', () => {
    const source = `
      /**
       * Old method for adding numbers
       * @deprecated Use add() instead
       * @param a - First number
       * @param b - Second number
       * @returns Sum
       */
      function oldAdd(a: number, b: number): number {
        return a + b;
      }
    `

    const docs = getTSDocs(source)
    expect(docs).toMatchSnapshot()
  })

  test('should handle optional parameters', () => {
    const source = `
      /**
       * Greets a person
       * @param name - Person's name
       * @param title? - Optional title
       * @returns Greeting message
       */
      function greet(name: string, title?: string): string {
        return title ? \`Hello \${title} \${name}\` : \`Hello \${name}\`;
      }
    `

    const docs = getTSDocs(source)
    expect(docs).toMatchSnapshot()
  })

  test('should handle functions without documentation', () => {
    const source = `
      function undocumented(x: number): number {
        return x * 2;
      }
    `

    const docs = getTSDocs(source)
    expect(docs).toEqual('- Comment\n  - Section')
  })
})
