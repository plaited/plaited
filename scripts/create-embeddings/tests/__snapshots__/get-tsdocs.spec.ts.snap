// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`getTSDocs should generate AST documentation structure 1`] = `
"- Comment
  - Section
    - Paragraph
      - PlainText
        * PlainText="Adds two numbers together"
      - SoftBreak
        * SoftBreak="\\n"
  - ParamCollection
    - ParamBlock
      - BlockTag
        * BlockTag="@param"
      * Spacing=" "
      * ParamBlock_ParameterName="a"
      * Spacing=" "
      * ParamBlock_Hyphen="-"
      * Spacing=" "
      - Section
        - Paragraph
          - PlainText
            * PlainText="First number"
          - SoftBreak
            * SoftBreak="\\n"
    - ParamBlock
      - BlockTag
        * BlockTag="@param"
      * Spacing=" "
      * ParamBlock_ParameterName="b"
      * Spacing=" "
      * ParamBlock_Hyphen="-"
      * Spacing=" "
      - Section
        - Paragraph
          - PlainText
            * PlainText="Second number"
          - SoftBreak
            * SoftBreak="\\n"
  - Block
    - BlockTag
      * BlockTag="@returns"
    - Section
      - Paragraph
        - PlainText
          * PlainText=" The sum of a and b"
        - SoftBreak
          * SoftBreak="\\n"
  - Block
    - BlockTag
      * BlockTag="@example"
    - Section
      - Paragraph
        - SoftBreak
          * SoftBreak="\\n"
      - FencedCode
        * FencedCode_OpeningFence="\`\`\`"
        * FencedCode_Language="ts"
        * Spacing="\\n"
        * FencedCode_Code="add(1, 2) // returns 3\\n"
        * Spacing=""
        * FencedCode_ClosingFence="\`\`\`"
        * Spacing="\\n""
`;

exports[`getTSDocs should handle functions with deprecated tag 1`] = `
"- Comment
  - Section
    - Paragraph
      - PlainText
        * PlainText="Old method for adding numbers"
      - SoftBreak
        * SoftBreak="\\n"
  - Block
    - BlockTag
      * BlockTag="@deprecated"
    - Section
      - Paragraph
        - PlainText
          * PlainText=" Use add() instead"
        - SoftBreak
          * SoftBreak="\\n"
  - ParamCollection
    - ParamBlock
      - BlockTag
        * BlockTag="@param"
      * Spacing=" "
      * ParamBlock_ParameterName="a"
      * Spacing=" "
      * ParamBlock_Hyphen="-"
      * Spacing=" "
      - Section
        - Paragraph
          - PlainText
            * PlainText="First number"
          - SoftBreak
            * SoftBreak="\\n"
    - ParamBlock
      - BlockTag
        * BlockTag="@param"
      * Spacing=" "
      * ParamBlock_ParameterName="b"
      * Spacing=" "
      * ParamBlock_Hyphen="-"
      * Spacing=" "
      - Section
        - Paragraph
          - PlainText
            * PlainText="Second number"
          - SoftBreak
            * SoftBreak="\\n"
  - Block
    - BlockTag
      * BlockTag="@returns"
    - Section
      - Paragraph
        - PlainText
          * PlainText=" Sum"
        - SoftBreak
          * SoftBreak="\\n""
`;

exports[`getTSDocs should handle optional parameters 1`] = `
"- Comment
  - Section
    - Paragraph
      - PlainText
        * PlainText="Greets a person"
      - SoftBreak
        * SoftBreak="\\n"
  - ParamCollection
    - ParamBlock
      - BlockTag
        * BlockTag="@param"
      * Spacing=" "
      * ParamBlock_ParameterName="name"
      * Spacing=" "
      * ParamBlock_Hyphen="-"
      * Spacing=" "
      - Section
        - Paragraph
          - PlainText
            * PlainText="Person's name"
          - SoftBreak
            * SoftBreak="\\n"
    - ParamBlock
      - BlockTag
        * BlockTag="@param"
      * Spacing=" "
      * ParamBlock_ParameterName="title"
      - Section
        - Paragraph
          - PlainText
            * PlainText="? - Optional title"
          - SoftBreak
            * SoftBreak="\\n"
  - Block
    - BlockTag
      * BlockTag="@returns"
    - Section
      - Paragraph
        - PlainText
          * PlainText=" Greeting message"
        - SoftBreak
          * SoftBreak="\\n""
`;
